lib/web_ui/flutter_js/src/canvaskit_loader.js

```diff
@@ -25,24 +25,11 @@ export const loadCanvasKit = (deps, config, browserEnvironment, canvasKitBaseUrl
       canvasKitUrl = deps.flutterTT.policy.createScriptURL(canvasKitUrl);
     }
     const wasmInstantiator = createWasmInstantiator(joinPathSegments(baseUrl, "canvaskit.wasm"));
-    const script = document.createElement("script");
-    script.src = canvasKitUrl;
-    if (config.nonce) {
-      script.nonce = config.nonce;
-    }
-    script.addEventListener("load", async () => {
-      try {
-        const canvasKit = await CanvasKitInit({
-          instantiateWasm: wasmInstantiator,
-        });
-        window.flutterCanvasKit = canvasKit;
-        resolve(canvasKit);  
-      } catch (e) {
-        reject(e);
-      }
+    const canvasKitModule = await import(canvasKitUrl);
+    window.flutterCanvasKit = await canvasKitModule.default({
+      instantiateWasm: wasmInstantiator,
     });
-    script.addEventListener("error", reject);
-    document.head.appendChild(script);
-  });
+    return window.flutterCanvasKit;
+  })();
   return window.flutterCanvasKitLoaded;
```

Comment:
I wonder if we need `flutterCanvasKitLoaded` anymore? I think we could simplify this to be `flutterCanvasKit` only?

This seems to be the loading code:

* https://github.com/flutter/engine/blob/e03c1c65f83fb0193b7148666d2977890c9af960/lib/web_ui/lib/src/engine/canvaskit/renderer.dart#L77-L86

(`windowFlutterCanvasKit` could be also `promiseToFuture`d and `await`ed just fine, couldn't it?)

((Not sure what happens when we await something that is not a Promise with promiseToFuture, though))

Reply:
`windowFlutterCanvasKitLoaded` is the thing we actually need, and is what is used in the fast path. But we still need `windowFlutterCanvasKit` because I think some people actually set that variable manually. I also think it is what is used on the hot reload path. (Maybe I'm having deja vu but I think we've had this exact discussion in a previous PR).

Reply:
> Maybe I'm having deja vu but I think we've had this exact discussion in a previous PR

Haha I wouldn't be surprised...  x'D